runtime/lib/ttnn/operations/pool/maxpool2d.cpp

```diff
@@ -7,6 +7,31 @@
 #include "tt/runtime/ttnn/operations/utils.h"
 
 namespace tt::runtime::ttnn::operations::pool {
+
+static ::ttnn::Tensor
+preshardForMaxPool2d(const ::ttnn::Tensor &input,
+                     const ::tt::target::ttnn::MaxPool2dOp *op,
+                     ::ttnn::Device &device) {
+  uint32_t output_height =
+      1 + (op->input_height() + 2 * op->padding_height() -
+           op->dilation_height() * (op->kernel_height() - 1) - 1) /
+              op->stride_height();
+  uint32_t output_width =
+      1 + (op->input_width() + 2 * op->padding_width() -
+           op->dilation_width() * (op->kernel_width() - 1) - 1) /
+              op->stride_width();
+
+  auto parallel_config =
+      ::ttnn::operations::conv::conv2d::determine_parallel_config(
```

Comment:
In the max pool op, they are already sharding the input... Can you explain what are you doing differently here? It is a good practice to leave these kinds of details in the commit message... Thanks!

Reply:
This presharding is copied right out of the TTNN maxpool operator python tests. In particular, without this addition Fatal errors occur when dim -2 is not tile-aligned.

Reply:
@pilkicTT If you check the related forge pr: https://github.com/tenstorrent/tt-forge-fe/pull/298 you will see that I've added a convolution to the output of the maxpool op. The tests all pass still.

Reply:
@nvukobratTT I've made a small change to the maxpool tests in forge to increase the scope. With options to place a producer and/or consumer conv. This covers the scenarios:

- conv -> maxpool -> conv
- conv -> maxpool
- maxpool ->conv

Draft PR: https://github.com/tenstorrent/tt-forge-fe/pull/304